src/main/java/com/fasterxml/jackson/core/JsonFactory.java

```diff
@@ -262,7 +262,7 @@ public static int collectDefaults() {
     /**
      * @since 2.16
      */
-    protected BufferRecyclerPool _bufferRecyclerPool;
+    protected BufferRecyclerPool<BufferRecycler> _bufferRecyclerPool;
```

Comment:
Will probably consider renaming of `BufferRecyclerPool` as it'll allow recycling other things, but initially leaving name as-is to simplify merging to `master`.